File: backend/src/main/java/com/davnig/invoicemanagementapi/model/entity/TaxableSubject.java
package com.davnig.invoicemanagementapi.model.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@NoArgsConstructor
@Table(name = "t_taxable_subject")
public class TaxableSubject {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "taxable-subject-id-gen")
    @SequenceGenerator(name = "taxable-subject-id-gen", sequenceName = "t_taxable_subject_id_seq", allocationSize = 1)
    @JsonIgnore
    private int id;
    @Column(name = "p_iva")
    private String piva;
    private String firstName;
    private String lastName;
    private String taxCode;

}

File: backend/src/main/kotlin/com/davnig/invoicemanagementapi/model/entity/TaxableSubject.kt
package com.davnig.invoicemanagementapi.model.entity

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.*

@Entity
@Table(name = "t_taxable_subject")
class TaxableSubject {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "taxable-subject-id-gen")
    @SequenceGenerator(name = "taxable-subject-id-gen", sequenceName = "t_taxable_subject_id_seq", allocationSize = 1)
    @JsonIgnore
    var id: Int = 0

    @Column(name = "p_iva")
    var piva: String? = null
    var firstName: String? = null
    var lastName: String? = null
    var taxCode: String? = null

}